CHANGELOG.md

```diff
@@ -9,6 +9,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Added `PipelineTask.add_observer()` to allow adding observers after task
```

Comment:
Maybe add a sentence like `Useful for cases where the task is passed around to other code components that need to add their own observers`.